docs/src/examples/components/Popup/Variations/PopupExamplePosition.shorthand.tsx

```diff
@@ -1,54 +1,91 @@
+const PopupExamplePosition = () => {
+  const [open] = useBooleanKnob({ name: 'open shorthand', initialValue: true })
 
-  return (
-    <Popup
-      align={align}
-      position={position}
-      trigger={<Button icon={icon} styles={{ padding, height: '38px', minWidth: '64px' }} />}
-      content={{
-        content: (
-          <p>
-            The popup is rendered {position} the trigger
-            <br />
-            aligned to the {align}.
-          </p>
-        ),
-      }}
-    />
+  const [position] = useSelectKnob<Position>({
+    name: 'position shorthand',
+    initialValue: 'above',
+    values: ['above', 'below', 'before', 'after'],
+  })
+
+  const [positionBeforeOrAfter, setPositionBeforeOrAfter] = React.useState<boolean>(
+    isPositionBeforeOrAfter(position),
   )
-}
 
-const triggers = [
-  { position: 'above', align: 'start', icon: 'arrow circle up', padding: '5px 42px 18px 5px' },
-  { position: 'above', align: 'center', icon: 'arrow circle up', padding: '5px 5px 18px 5px' },
-  { position: 'above', align: 'end', icon: 'arrow circle up', padding: '5px 5px 18px 42px' },
-  { position: 'below', align: 'start', icon: 'arrow circle down', padding: '18px 42px 5px 5px' },
-  { position: 'below', align: 'center', icon: 'arrow circle down', padding: '18px 5px 5px 5px' },
-  { position: 'below', align: 'end', icon: 'arrow circle down', padding: '18px 5px 5px 42px' },
-  { position: 'before', align: 'top', icon: 'arrow circle left', padding: '5px 42px 18px 5px' },
-  { position: 'before', align: 'center', icon: 'arrow circle left', padding: '5px 42px 5px 5px' },
-  { position: 'before', align: 'bottom', icon: 'arrow circle left', padding: '18px 42px 5px 5px' },
-  { position: 'after', align: 'top', icon: 'arrow circle right', padding: '5px 5px 18px 42px' },
-  { position: 'after', align: 'center', icon: 'arrow circle right', padding: '5px 5px 5px 42px' },
-  { position: 'after', align: 'bottom', icon: 'arrow circle right', padding: '18px 5px 5px 42px' },
-]
-
-const PopupExamplePosition = () => (
-  <Grid columns="repeat(3, 30px)" variables={{ padding: '30px', gridGap: '80px' }}>
-    {triggers.map(({ position, align, icon, padding }) => (
-      <PopupWithButton
-        position={position}
+  const [align] = useSelectKnob<Alignment>(
+    positionBeforeOrAfter
+      ? {
+          name: 'h-align shorthand',
+          initialValue: 'top',
+          values: ['top', 'center', 'bottom'],
+        }
+      : {
+          name: 'v-align shorthand',
+          initialValue: 'start',
+          values: ['start', 'center', 'end'],
+        },
+  )
+
+  React.useEffect(() => setPositionBeforeOrAfter(isPositionBeforeOrAfter(position)), [position])
+
+  const buttonStyles = { padding: paddings[position][align], height: '38px', minWidth: '64px' }
+
+  return (
+    <Grid columns="1" variables={{ padding: '100px 0' }} styles={{ justifyItems: 'center' }}>
+      <Popup
+        pointing
+        open={open}
```

Comment:
This means that if the checkbox is not checked, it is not possible to open the popup by clicking the trigger.
I would use something like `open={open || undefined}`

Reply:
We can also use `onOpenChange`, each knob contains a setter in its tuple: `const [open, setOpen] = useBooleanKnob()`

Reply:
It'd be a nice thing to add. It worked for the `Popup` but didn't for the `Dropdown`. When I deselect the `open` checkbox, Downshift starts updating the current state of `open` flag before the `open` prop is changed from `true` to `false`.

Order of events is strange.. I debugged this with @layershifter all day and didn't reach a conclusion :(

Any suggestions @miroslavstastny ? I'd keep this as is until we have issues impacting users.